spec/factories/transaction_types.rb

```diff
@@ -12,16 +12,17 @@ def repeating_sequence_for_transaction_type(type, current)
 
 FactoryBot.define do
   factory :transaction_type do
-    name { TransactionType::NAMES.values.flatten.sample }
-    operation { TransactionType::NAMES.keys.sample }
```

Comment:
We want our tests to be deterministic, so I don't think we should be doing things like `.sample` in our factories (or using Faker!)

Reply:
Generally agree that non-detereministic test data is a pain - probably worthy of a wider discussion as limited non-deterministic test data, over time, can identify bugs that a deterministic will not. overall 👍